### ISSUE TITLE
systemd module not enabling service if already enabled in sysv

### ISSUE THREAD

**Issue by gaudenz** [ansible/ansible#22303]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
systemd module

##### ANSIBLE VERSION
```
ansible 2.2.2.0
```

##### CONFIGURATION
nothing changed which could influence this

##### OS / ENVIRONMENT
ansible host: Debian testing
target host: Debian stable

##### SUMMARY
```yaml
systemd:
  name: myservice
  enabled: True
```
Does not enable `myservice` if there is an enabled sysv init script with the same name. 

##### STEPS TO REPRODUCE
Using exabgp (from backports) as an example package which has both a systemd service file and an init script. The systemd service is not enabled after installation while the sysv init is enabled. Many other packages are affected as well. 

```yaml
- name: Install ExaBGP
  apt:
    name: exabgp
    state: installed
- name: Enable Service
  systemd:
    name: exabgp
    enabled: True
```

The following playbook also does not work, because the `service` task actually does not use the service module but the systemd module if it detects an OS using systemd.

```yaml
- name: Install ExaBGP
  apt:
    name: exabgp
    state: installed
- name: Enable Service
  service:
    name: exabgp
    enabled: True
```

The following playbook works around the issue by explicitly using the service module instead of the systemd module. The service module had the same issue fixed in ansible/ansible-modules-core#3764.

```yaml
- name: Install ExaBGP
  apt:
    name: exabgp
    state: installed
- name: Enable Service
  service:
    name: exabgp
    enabled: True
    use: service
```

The root cause is probably this if clause in the systemd module:
```
        elif rc == 1:
            # if not a user service and both init script and unit file exist stdout should have enabled/disabled, otherwise use rc entries
            if not module.params['user'] and \
               is_initd and \
               (not out.strip().endswith('disabled') or sysv_is_enabled(unit)):

                enabled = True
```
The `or`part at the end marks any service enabled in sysv as also enabled in systemd.

##### EXPECTED RESULTS
The exabgp systemd service should be enabled after running this script.

##### ACTUAL RESULTS
The service is not enabled.

**Comment by bcoca** [comment]
isn't it a problem if both are enabled? it could cause double execution of the same daemon.

**Comment by gaudenz** [comment]
No this is not possible as you can't run sysv init and systemd at the same time on the same host. And for systemd's sysv compatibility layer a service unit of the same name masks an init script. Systemd won't run the init script if both are enabled.

**Comment by geerlingguy** [comment]
Commenting that my PHP role is impacted by this, but only on Debian 8 Jessie. It seems that some packages place both an initv script and systemd unit on the system.

(Also noting that I'm seeing this behavior in 2.3.0)

**Comment by gaudenz** [comment]
At least for Debian jessie and stretch having a sysvinit script and (optionally) a systemd unit in packages shipping a daemon is mandatory. This is also true for most Ubuntu packages. At least for now Debian wants to retain sysv support although systemd is the default init system.

**Comment by stephankn** [comment]
This problem can still be observed with `ansible 2.4.0 (devel 5f23aa69c6)` and trying to enable zabbix-server on Ubuntu 16.04 LTS.

**Comment by geerlingguy** [comment]
Ugh, hit this on another server build with Varnish (default Varnish package repos include initv and systemd configuration) today. I was using an older version of my role which still had the plain `service` usage, but it was fixed in the PR above.

It would save a lot of debugging time to be able to see more easily what 'service' command/system is being used for a given playbook run, or maybe have some sort of host var we could set that says 'don't use sysvinit' or something like that.

**Comment by bcoca** [comment]
@geerlingguy if running under debug all 'shell outs' are logged on the target

From docs  systemd should executing the service when its sysv enabled. does this pose a problem i'm not aware of? 
Debian Jessie has a broken systemd implementation, which is why this code handles it this way, I'm not against changing it but would like to do in a way that it is workable for all.

**Comment by geerlingguy** [comment]
In this case it was on Ubuntu 16.04; I haven't had time to dig into the code yet, but if I hit this again, I will likely do so, just to try to find a way to get it fixed long-term. I haven't had any issues on Debian 9 or CentOS 7, so YMMV.

**Comment by Adri2000** [comment]
@bcoca that's not how I understand https://www.freedesktop.org/wiki/Software/systemd/FrequentlyAskedQuestions/ - one of the answers there explicitly says that a sysv script that is enabled does not mean the corresponding unit is enabled, or that the sysv init script will be executed.

**Comment by KevDBG** [comment]
Hello,

Same problem with Ubuntu 16.04 for service td-agent/Fluent.

The workaround is to use "use: service"
```

- name: Enable and start td-agent
  become: yes
  become_method: sudo
  service:
    name: td-agent
    enabled: yes
    state: started
    use: service
```

It would be interesting to use systemctl and not service.

**Comment by ThomasSteinbach** [comment]
Thank you @Zophren - `use: service` solved this issue for me in Ubuntu 18.04.
It could have been so simple, but it isn't. Maybe I hate computers.

**Comment by resmo** [comment]
@gaudenz could you verify, if #46245 solves the problem?
